packages/edit-post/src/editor.js

```diff
@@ -64,23 +64,34 @@ function Editor( {
 			getEntityRecords,
 		} = select( 'core' );
 		const { getEditorSettings, getCurrentPost } = select( 'core/editor' );
+		const currentPostObj = getCurrentPost();
```

Comment:
The EditorProvider component rendered here is the  component responsible for  setting up the  current post in state, this means for me that  we  shouldn't be using the "currentPost" to actually fetch the "post" to be passed too EditorProvider, it creates a cycle.

Reply:
This might not be the right approach but we spent a faire amount of time with @jrtashjian to try to understand why `setupEditor` action wasn't behaving like WordPress 5.6 (see video in description). When you use setupEditor with a new post entity it actually sets the current post ID (and the URL as well) for the editor.  The problem is that it doesn't sets the reference _everywhere_. Here we can see this component was referencing the initial post id from when the page loaded and thus creating all sorts of oddities. 

The use case being to be able to have the Editor load another post or page completely without having to go back to the admin. We found setupEditor from the 'core/editor' store to work pretty well for our usecase but it was broken in the latest version of Gutenberg. This actually seems to fix all the problem we have been seeing so far.

Reply:
Judging by the end2end test failures, this is far from the right  way to do this IMO. In fact I don't think being able to change the "id", and "postType" of the edit-post package is something that was even considered as a  use-case for setupEditor.

edit-post depend on editor and not the other way around, meaning that it's edit-post that pass "data"/"config"   to editor package. This means  that  changing the state  of  "editor" by calling "setupEditor" for instance won't have any impact on edit-post. (and a recent refactor to edit-post might have made this  more visible).

I  think if you  want to  switch the edited post without leaving the  page,  you  should instead consider using ` wp.editPost.reinitializeEditor` (I didn't try but it seems more logical to me at least)

Reply:
@youknowriad we did try `wp.editPost.reinitializeEditor` but it doesn't work unfortunately. And fixing the reinitializeEditor function to have the right intent was unmounting and remounting everything thus creating a "blip" in the browser. This was honestly the least intrusive way we could find while keeping the experience smooth (sidebars and position all stayed in place and only the content is changing). 

I totally understand the cycle effect though but fixing that is way more involved and that is way I was proposing something simple as a stepping stone just to make sure we are backward compatible to wp/5.6.

Reply:
From my testing, since most thing are already heavily tied to the core store entity to update themselves it is quite fast to change the current post ID reference that way. Is there a reason why components in edit-post are not more heavily reliant on a redux store instead of a props for the current post id and post type?